### ISSUE TITLE
Jest is automatically in scope. Do not import "jest" in Config file.

### ISSUE THREAD

**Issue by SalahAdDin** [testing-library/eslint-plugin-jest-dom#289]
- `eslint-plugin-jest-dom` version: 4.0.3
- `node` version: v18.12.1
- `npm` version: 8.19.2

Relevant code or config

```typescript
import type { Config } from "jest";

const customJestConfig: Config = {
  preset: "ts-jest",
  // We want to clear mocks and instances between every st
  clearMocks: true,
```

What you did: Just set up the config file.

What happened: 
The linter complains about the importing line.
![image](https://user-images.githubusercontent.com/5034215/214025768-29440da6-d1fd-4185-b683-b6aafb5d7d55.png)

Reproduction repository:

Problem description:
The documentation says [we need to import the correct type](https://jestjs.io/docs/next/configuration) for the configuration object if we are working with **Typescript.**

However, when we import it, the linter complains about it.

Suggested solution:
The linter should ignore the configuration file, maybe...

**Comment by G-Rath** [comment]
Duplicate of https://github.com/jest-community/eslint-plugin-jest/issues/1333

**Comment by SalahAdDin** [comment]
@G-Rath so, it will be automatically fixed in a new release version, right?

**Comment by G-Rath** [comment]
@SalahAdDin this has nothing to do with this plugin, so no

**Comment by SalahAdDin** [comment]
> @SalahAdDin this has nothing to do with this plugin, so no

It is related to the main package, wich also deprecated the rule; so, if this package updates the package, the big will be gone.

**Comment by G-Rath** [comment]
This package does not expose that rule as its own, so there is no bug - there's actually not a bug in the base rule to begin with becuase it's not meant to be run on jest config (which is outlined in the overrides example in our readme)

**Comment by SalahAdDin** [comment]
> This package does not expose that rule as its own, so there is no bug - there's actually not a bug in the base rule to begin with because it's not meant to be run on jest config (which is outlined in the overrides example in our readme)

No idea, I just moved to `vitest`. I followed the documentation and enabled the linter, it is being used also in the config file.
